Platon/backend/app/workspace_system/views.py

```diff
@@ -0,0 +1,174 @@
+from flask import current_app as app
+from flask_restplus import Namespace, Resource
+from flask import make_response,jsonify,request
+
+from app import api, db
+
+from app.workspace_system.forms import CreateWorkspaceForm, create_workspace_parser
+from app.workspace_system.forms import ReadWorkspaceForm, read_workspace_parser
+from app.workspace_system.forms import UpdateWorkspaceForm, update_workspace_parser
+from app.workspace_system.forms import DeleteWorkspaceForm, delete_workspace_parser
+
+from app.workspace_system.models import Workspace, WorkspaceSkill, WorkspaceRequirement, Contribution, Requirement
+from app.profile_management.models import Skills
+
+from app.auth_system.helpers import login_required
+from app.workspace_system.helpers import *
+from app.file_system.helpers.FileSystem import initialize_ws
+
+workspace_system_ns = Namespace("Workspace System",
+                            description="Workspace System Endpoints",
+                            path = "/workspaces")
+
+
+@workspace_system_ns.route("")
+class WorkspacesAPI(Resource):
+	# POST request
+    @api.expect(create_workspace_parser)
+    @api.doc(responses={
+                201: "Workspace has been successfully created.",
+                400: "Missing data fields or invalid data.",
+                500: "The server is not connected to the database."
+            })
+    @login_required
+    def post(requester_id, self):
+        '''
+        Creates a new workspace.
+        '''
+
+        # Parses the form data.
+        form = CreateWorkspaceForm(request.form)
+        if form.validate():
+            new_workspace_information = {**form.data}
+            new_workspace_skills = eval(new_workspace_information.pop("skills", "None"))
+            new_workspace_requirements = eval(new_workspace_information.pop("requirements", "None"))
+
+            new_workspace = Workspace(creator_id=requester_id, **new_workspace_information)
+            try:
+                db.session.add(new_workspace)
+                db.session.commit()
+            except:
+                return make_response(jsonify({"error" : "The server is not connected to the database. (Workspace could not get created.)"}, 500))
+            else:
+                initialize_ws(new_workspace.id)
+                try:
+                    add_workspace_skills(new_workspace.id, new_workspace_skills)
+                    add_workspace_requirements(new_workspace.id, new_workspace_requirements)
+                    add_workspace_contribution(new_workspace.id, requester_id)
```

Comment:
Can you create another endpoint to add a contribution? Because, in all of its use cases, to add a contribution to the workspace, an application or invitation must be made. So I think you have to remove it from here. There must be a different class that serves as endpoints to create, delete invitations and applications.

Reply:
@HUmutOzdemir I understand your point; but these helpers were not implemented to replace those endpoints, these here were only implemented to shorten the code of "Create Workspace" endpoint. I'll implement the endpoints regarding invitations and applications soon.